packages/pools/src/router/__tests__/routes.spec.ts

```diff
@@ -1,412 +1,1137 @@
-import { Int } from "@keplr-wallet/unit";
+import { Dec, Int } from "@keplr-wallet/unit";
 
+import { NotEnoughLiquidityError } from "../../errors";
+import { Route, RouteWithInAmount } from "../route";
+import { RoutablePool } from "../types";
 import {
   allPools,
   makeDefaultTestRouterParams,
+  makeRouterWithForceRoutes,
   makeStablePool,
   makeWeightedPool,
 } from "./pool";
 
 describe("OptimizedRoutes", () => {
-  describe("Simple: picks higher liquidity routes", () => {
-    test("picks higher liquidity pool", async () => {
-      const pools = [
-        makeWeightedPool({
-          id: "2",
-          firstPoolAsset: { amount: "1000" },
-          secondPoolAsset: { amount: "1000" },
-        }),
-        makeWeightedPool(),
-      ];
-
-      const router = makeDefaultTestRouterParams({
-        pools,
-        incentivizedPoolIds: [],
-        stakeCurrencyMinDenom: "ufoo",
+  describe("getOptimizedRoutesByTokenIn", () => {
+    describe("handles", () => {
+      it("no pools", () => {
+        const router = makeDefaultTestRouterParams({ pools: [] });
+        const promisedSplit = router.getOptimizedRoutesByTokenIn(
+          { denom: "uion", amount: new Int("100") },
+          "uosmo"
+        );
+        expect(promisedSplit).resolves.toEqual([]);
       });
-
-      const routes = await router.getOptimizedRoutesByTokenIn(
+      it("invalid tokens", () => {
```

Comment:
I also think that `it` statements should be read as full sentences to understand what the test is doing without having to run it within jest.